build: cmake 3.11.0 does not work with our makefile

This is on OSX 10.13.3 using brew.

Brew now installs and updates cmake to 3.11.0 and this is not compatible with out Makefile.

`make` will build librocksdb correctly, but then stop with the following (without an error):
```bash
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: librocksdb.a(db_impl_debug.cc.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: librocksdb.a(db_bench_tool.cc.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: librocksdb.a(rocks_lua_compaction_filter.cc.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: librocksdb.a(db_impl_debug.cc.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: librocksdb.a(db_bench_tool.cc.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: librocksdb.a(rocks_lua_compaction_filter.cc.o) has no symbols
```

If you run `make` again:
```bash
GOPATH set to /Users/bram/go
Running make with -j8
go run pkg/cmd/docgen/{main,funcs}.go functions docs/generated/sql --quiet
make[1]: /usr/local/Cellar/cmake/3.10.3/bin/cmake: No such file or directory
make[1]: *** [cmake_check_build_system] Error 1
make: *** [libroachccl] Error 2
make: *** Waiting for unfinished jobs....
touch bin/.docgen_functions
```

To downgrade back to cmake 3.10.3, run the following commands:
1. `brew unlink cmake`
1. `brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/a169609a5dcd5bf0cb83c2d8ffc7da934b3d81fe/Formula/cmake.rb`
1. `brew pin cmake`

Once this is fixed, you should be able to run `brew unpin cmake` and allow it to upgrade like normal.

--- a/Makefile
+++ b/Makefile
@@ -1282,6 +1282,7 @@ clean-c-deps:
 	rm -rf $(PROTOBUF_DIR)
 	rm -rf $(ROCKSDB_DIR)
 	rm -rf $(SNAPPY_DIR)
+	rm -rf $(LIBROACH_DIR)
 
 .PHONY: unsafe-clean-c-deps
 unsafe-clean-c-deps:
@@ -1290,6 +1291,7 @@ unsafe-clean-c-deps:
 	git -C $(PROTOBUF_SRC_DIR) clean -dxf
 	git -C $(ROCKSDB_SRC_DIR)  clean -dxf
 	git -C $(SNAPPY_SRC_DIR)   clean -dxf
+	git -C $(LIBROACH_SRC_DIR) clean -dxf
 
 .PHONY: clean
 clean: ## Remove build artifacts.

